stdlib/2/subprocess.pyi

```diff
@@ -2,65 +2,73 @@
 
 # Based on http://docs.python.org/2/library/subprocess.html and Python 3 stub
 
-from typing import Sequence, Any, AnyStr, Mapping, Callable, Tuple, IO, Union, Optional
+from typing import Sequence, Any, AnyStr, Mapping, Callable, Tuple, IO, Union, Optional, List, Text
 
 _FILE = Union[int, IO[Any]]
+_TXT = Union[bytes, Text]
+_CMD = Union[_TXT, Sequence[_TXT]]
 
 # Same args as Popen.__init__
-def call(args: Union[str, Sequence[str]],
+def call(args: _CMD,
          bufsize: int = ...,
-         executable: str = ...,
+         executable: _TXT = ...,
          stdin: _FILE = ...,
          stdout: _FILE = ...,
          stderr: _FILE = ...,
          preexec_fn: Callable[[], Any] = ...,
          close_fds: bool = ...,
          shell: bool = ...,
-         cwd: str = ...,
-         env: Mapping[str, str] = ...,
+         cwd: _TXT = ...,
+         env: Mapping[_TXT, _TXT] = ...,
          universal_newlines: bool = ...,
          startupinfo: Any = ...,
          creationflags: int = ...) -> int: ...
 
-def check_call(args: Union[str, Sequence[str]],
+def check_call(args: _CMD,
                bufsize: int = ...,
-               executable: str = ...,
+               executable: _TXT = ...,
                stdin: _FILE = ...,
                stdout: _FILE = ...,
                stderr: _FILE = ...,
                preexec_fn: Callable[[], Any] = ...,
                close_fds: bool = ...,
                shell: bool = ...,
-               cwd: str = ...,
-               env: Mapping[str, str] = ...,
+               cwd: _TXT = ...,
+               env: Mapping[_TXT, _TXT] = ...,
                universal_newlines: bool = ...,
                startupinfo: Any = ...,
                creationflags: int = ...) -> int: ...
 
 # Same args as Popen.__init__ except for stdout
-def check_output(args: Union[str, Sequence[str]],
+def check_output(args: _CMD,
                  bufsize: int = ...,
-                 executable: str = ...,
+                 executable: _TXT = ...,
                  stdin: _FILE = ...,
                  stderr: _FILE = ...,
                  preexec_fn: Callable[[], Any] = ...,
                  close_fds: bool = ...,
                  shell: bool = ...,
-                 cwd: str = ...,
-                 env: Mapping[str, str] = ...,
+                 cwd: _TXT = ...,
+                 env: Mapping[_TXT, _TXT] = ...,
                  universal_newlines: bool = ...,
                  startupinfo: Any = ...,
-                 creationflags: int = ...) -> str: ...
+                 creationflags: int = ...
+                 ) -> Any: ...  # morally: _TXT
```

Comment:
Can it really return unicode in py2? It returns str for me even with universal_newlines on. (In Python 3 it's bytes without universal_newlines and str with it.)